File: tests/fixtures/_highlight-corpus/sample.dart
// Dart sample
import 'dart:io';

class Counter {
  int _value = 0;
  int increment([int n = 1]) {
    _value += n;
    return _value;
  }
  int get value => _value;
}

Future<void> main(List<String> args) async {
  final file = File(args.isNotEmpty ? args[0] : 'input.txt');
  if (!await file.exists()) {
    stderr.writeln('input file not found');
    exit(1);
  }
  final counter = Counter();
  await for (final line in file
      .openRead()
      .transform(const SystemEncoding().decoder)
      .transform(const LineSplitter())) {
    if (line.isNotEmpty) {
      counter.increment(line.length);
    }
  }
  print('total bytes: ${counter.value}');
}

File: tests/fixtures/_highlight-corpus/sample.ml
(* OCaml sample *)
module Counter = struct
  type t = { mutable value : int }

  let make () = { value = 0 }

  let increment c n =
    c.value <- c.value + n;
    c.value

  let value c = c.value
end

let read_lines path =
  let ic = open_in path in
  let lines = ref [] in
  try
    while true do
      lines := input_line ic :: !lines
    done;
    assert false
  with End_of_file ->
    close_in ic;
    List.rev !lines

let () =
  let counter = Counter.make () in
  let lines = read_lines "input.txt" in
  List.iter
    (fun line ->
      if String.length line > 0 then
        ignore (Counter.increment counter (String.length line)))
    lines;
  Printf.printf "total bytes: %d\n" (Counter.value counter)
